src/types/debt_registry_entry.ts

```diff
@@ -0,0 +1,69 @@
+import { BigNumber } from "../../utils/bignumber";
+import { Web3Utils } from "../../utils/web3_utils";
+
+/*
+This type represents the data as received from calls to the `DebtRegistry`'s
+`get` method -- whose signature is included below.
+
+This data represents all the information we store on-chain for a given entry in
+the `DebtRegistry`.
+
+```
+function get(bytes32 issuanceHash)
+        public
+        view
+        returns(address, address, address, uint, address, bytes32, uint);
+```
+ */
+export type DebtRegistryEntryData = [string, string, string, BigNumber, string, string, BigNumber];
+
+/*
+This type maps 1:1 to the `Entry` struct in our `DebtRegistry` smart contract:
+
+```
+struct Entry {
+    address version;
+    address beneficiary;
+    address underwriter;
+    uint underwriterRiskRating;
+    address termsContract;
+    bytes32 termsContractParameters;
+    uint issuanceBlockTimestamp;
+}
+```
+ */
+export class DebtRegistryEntry {
+    constructor(
+        public version: string,
+        public beneficiary: string,
+        public underwriter: string,
+        public underwriterRiskRating: BigNumber,
+        public termsContract: string,
+        public termsContractParameters: string,
+        public issuanceBlockTimestamp: BigNumber,
+    ) {
+        this.version = version;
+        this.beneficiary = beneficiary;
+        this.underwriter = underwriter;
+        this.underwriterRiskRating = underwriterRiskRating;
+        this.termsContract = termsContract;
+        this.termsContractParameters = termsContractParameters;
+        this.issuanceBlockTimestamp = issuanceBlockTimestamp;
+    }
+
+    public getHash(): string {
```

Comment:
There's never really a time when we need the hash of the `DebtRegistryEntry`, so we can remove this.